packages/celltags-extension/src/celltag.tsx

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) Jupyter Development Team.
+ * Distributed under the terms of the Modified BSD License.
+ */
+
+import React from 'react';
+import { FieldProps } from '@rjsf/utils';
+import { INotebookTracker } from '@jupyterlab/notebook';
+import { addIcon, checkIcon, LabIcon } from '@jupyterlab/ui-components';
+import { reduce } from '@lumino/algorithm';
+import {
+  ITranslator,
+  nullTranslator,
+  TranslationBundle
+} from '@jupyterlab/translation';
+
+export class CellTagField {
+  constructor(tracker: INotebookTracker, translator?: ITranslator) {
+    this._tracker = tracker;
+    this._translator = translator || nullTranslator;
+    this._trans = this._translator.load('jupyterlab');
+    this._editing = false;
+  }
+
+  addTag(props: FieldProps, tag: string) {
+    const data = props.formData;
+    if (tag && !data.includes(tag)) {
+      data.push(tag);
+      props.formContext.updateMetadata({ [props.name]: data }, true);
+    }
+  }
+
+  /**
+   * Pull from cell metadata all the tags used in the notebook and update the
+   * stored tag list.
+   */
+  pullTags(): string[] {
+    const notebook = this._tracker?.currentWidget;
+    const cells = notebook?.model?.cells;
+    if (cells === undefined) {
+      return [];
+    }
+    const allTags = reduce(
+      cells,
+      (allTags: string[], cell) => {
+        const tags: string[] = (cell.getMetadata('tags') as string[]) ?? [];
+        return [...allTags, ...tags];
+      },
+      []
+    );
+    return [...new Set(allTags)].filter(tag => tag !== '');
+  }
+
+  private _emptyAddTag(target: HTMLInputElement) {
+    target.value = '';
+    target.style.width = '49px';
```

Comment:
Should either css variable or relative length unit like `em` be used here to make sure width of the element is responsive (to for example work area or viewport width changes)?

Should we opt for class names + css sheets vs inline styling? CC @krassowski

Reply:
Sure, I'll add a class for empty input.

Reply:
Added a `jp-CellTags-Empty` class and replaced the `width: '49px'` by a `width: '4em'` in css sheets.

Reply:
Thank you @brichet